dashboard/db/migrate/20160818190145_add_unique_constraint_to_pd_workshop_attendance.rb

```diff
@@ -0,0 +1,24 @@
+class AddUniqueConstraintToPdWorkshopAttendance < ActiveRecord::Migration
+  def up
+    duplicate_attendance_values = Pd::Attendance.
+      select('pd_session_id, teacher_id, count(*)').
+      group(:pd_session_id, :teacher_id).
+      having('count(*) > 1').
+      pluck(:pd_session_id, :teacher_id)
+
+    # Destroy all but the latest of each set
+    duplicate_attendance_values.each do |pd_session_id, teacher_id|
+      Pd::Attendance.
+        where(pd_session_id: pd_session_id, teacher_id: teacher_id).
+        order(id: :desc)[1..-1].
+        each(&:destroy)
+    end
+
+    # Add unique multi-column index
+    add_index :pd_attendances, [:pd_session_id, :teacher_id], unique: true
+  end
```

Comment:
There seems to be a pre-existing index on `pd_session_id`. Presumably this should be removed (in the `up`) if we are indexing on `(pd_session_id, teacher_id)` and added (in the `down`)?